src/main/java/edu/harvard/iq/dataverse/dataaccess/DataConverter.java

```diff
@@ -220,28 +218,50 @@ private static File runFormatConversion (DataFile file, File tabFile, String for
             return tabFile;
         }
 
-        File formatConvertedFile = null;
+        File formatConvertedFile;
         // create the service instance
         RemoteDataFrameService dfs = new RemoteDataFrameService();
         
+        List<String> directFormats = Arrays.asList(new String[]{"application/x-stata",
+                                                                "application/x-spss-sav",
+                                                                "application/x-spss-por"});
         if ("RData".equals(formatRequested)) {
-            List<DataVariable> dataVariables = file.getDataTable().getDataVariables();
-            Map<String, Map<String, String>> vls = null;
-
-            vls = getValueTableForRequestedVariables(dataVariables);
-            logger.fine("format conversion: variables(getDataVariableForRequest())=" + dataVariables + "\n");
-            logger.fine("format conversion: variables(dataVariables)=" + dataVariables + "\n");
-            logger.fine("format conversion: value table(vls)=" + vls + "\n");
-            RJobRequest sro = new RJobRequest(dataVariables, vls);
-
-            sro.setTabularDataFileName(tabFile.getAbsolutePath());
-            sro.setRequestType(SERVICE_REQUEST_CONVERT);
-            sro.setFormatRequested(FILE_TYPE_RDATA);
-
-        
-
-            // execute the service
-            Map<String, String> resultInfo = dfs.execute(sro);
+            String origFormat = file.getOriginalFileFormat();
+            Map<String, String> resultInfo;
+            if (origFormat.contains("stata") || origFormat.contains("spss")){
+                if (origFormat.contains("stata")){
+                    origFormat = "dta";
+                } else if (origFormat.contains("sav")){
+                    origFormat = "sav";
+                } else if (origFormat.contains("por")){
+                    origFormat = "por";
+                }
+                    
+                try {
+                    StorageIO<DataFile> storageIO = file.getStorageIO();
+                    long size = storageIO.getAuxObjectSize("orig");
+                    File origFile = downloadFromByteChannel((ReadableByteChannel) storageIO.openAuxChannel("orig"), size);
```

Comment:
I don't know whether this cast to `ReadableByteChannel` should instead be a change to `StorageIO`, as all current ones return `ReadableByteChannel`s. I figured that that is a bigger change, and probably needs an OK from @landreev and/or @scolapasta  first.